File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution{
    private int count = 0;
    public void dfs(TreeNode root , int maxele){
        if(root == null) return; 
        if(root.val >= maxele){
            count++;
            maxele = root.val;
        }
        dfs(root.left, maxele);
        dfs(root.right, maxele);
    }
    public int goodNodes(TreeNode root) {
        
        int maxele = Integer.MIN_VALUE;
        dfs(root , maxele);
        return count;
    }
}

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root , int &count , int tillMaxele){
        if(root==NULL) return;
        if(root->val >= tillMaxele){
            count++;
            tillMaxele = root->val;
        }
        dfs(root->left , count , tillMaxele);
        dfs(root->right, count, tillMaxele);

    }
    int goodNodes(TreeNode* root) {
        int count = 0;
        int tillMaxele = INT_MIN;
        dfs(root , count , tillMaxele);
        return count;
        
    }
};